FindMyCoffee.playground/Sources/FindMyCoffeApp.swift

```diff
@@ -0,0 +1,54 @@
+
+
+public class FindMyCoffeeApp {
+    
+    private var userLocation:Coordinates!
+    private var coffeeShopsDatabase:String!
+
+    lazy var availableCoffeeShops:[CoffeeShop]? = {
+        return parseCoffeeShopDatabase()
+    }()
+    
+    // MARK:- Constructor
+    
+    public init(_ userLocation:Coordinates, coffeeShopsDatabase:String) {
+        self.userLocation       = userLocation
+        self.coffeeShopsDatabase = coffeeShopsDatabase
+    }
+    
+    // MARK:- Public methods
+    
+    public func findSolution() -> [CoffeeShop] {
```

Comment:
Yes, you are right. I had in mind the possibility of getting  the solution without calling the showSolution() method, in case I needed to process them in a different way (like parse in another format)